cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterContractTypeCreateController.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public class WaterContractTypeCreateController implements Handler {
+    public static final String TAG = "Water Contracts";
+    private final MetricRegistry metrics;
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public WaterContractTypeCreateController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected WaterContractDao getContractDao(DSLContext dsl) {
+        return new WaterContractDao(dsl);
+    }
+
+    @OpenApi(
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = LookupType.class, type = Formats.JSONV1)
+            },
+            required = true),
+        queryParams = {
+            @OpenApiParam(name = FAIL_IF_EXISTS, type = boolean.class, description = "Create will fail if provided"
+                    + "display value already exists. Default: true")
+        },
+        pathParams = {
+            @OpenApiParam(name = NAME, description = "The name of the contract.", required = true),
+            @OpenApiParam(name = OFFICE, description = "The office Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The project Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The water user the contract is associated with.",
+                    required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = "204", description = "Contract type successfully stored to CWMS."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Create a new water contract type",
+        method = HttpMethod.POST,
+        path = "/projects/{office}/{project-id}/water-users/{water-user}/contracts/{name}/types",
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(CREATE)) {
+            DSLContext dsl = getDslContext(ctx);
+            boolean failIfExists = ctx.queryParam(FAIL_IF_EXISTS) == null || Boolean.parseBoolean(ctx
+                    .queryParam(FAIL_IF_EXISTS));
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, LookupType.class);
+            ctx.contentType(contentType.toString());
+            LookupType contractType = Formats.parseContent(contentType, ctx.body(), LookupType.class);
+            List<LookupType> types = new ArrayList<>();
```

Comment:
Why add this to an array rather than have the dao take in a single lookuptype?

Reply:
Swapped DAO to take single lookuptype